autofaiss/indices/distributed.py

```diff
@@ -644,7 +646,8 @@ def _create_small_indexes(embedding_root_dirs: List[str]) -> List[Optional[Dict[
 
     # Create small and big indexes
     all_metrics = []
-    with ThreadPool(256) as p:
+    n_threads = min(maximum_nb_threads, len(big_partitions) + int(len(small_partitions) > 0))
```

Comment:
If `partitions` is empty, `n_threads` will be 0, we can handle this edge case at the begging of this function.